tracer/test/Datadog.Trace.IntegrationTests/ContainerTaggingTests.cs

```diff
@@ -52,15 +39,13 @@ public async Task Http_Headers_Contain_ContainerId()
 
                 await tracer.FlushAsync();
 
-                var spans = agent.WaitForSpans(1);
-                Assert.Equal(1, spans.Count);
-                Assert.Equal(expectedContainedId, actualContainerId);
+                var spans = agent.WaitForSpans(count: 1);
+                Assert.Equal(expected: 1, spans.Count);
 
-                if (EnvironmentTools.IsWindows())
-                {
-                    // we don't extract the containerId on Windows (yet?)
-                    Assert.Null(actualContainerId);
-                }
+                // we don't extract the containerId on Windows (yet?)
+                Assert.Equal(expected: 1, agent.TraceRequestHeaders.Count);
+                var expectedContainedId = EnvironmentTools.IsWindows() ? null : ContainerMetadata.GetContainerId();
+                Assert.All(agent.TraceRequestHeaders, headers => Assert.Equal(expectedContainedId, headers[AgentHttpHeaderNames.ContainerId]));
```

Comment:
This looks like it checks all the headers, but we only want to check `AgentHttpHeaderNames.ContainerId`. Won't there be additional headers here that we should ignore?

Reply:
Yeah, I think we could/should use fluent assertions to make this easier: 
```suggestion
                var headers = agent.RequestHeaders.Should().ContainSingle().Subject;
                headers.AllKeys.ToDictionary(x=> x, x=> headers[x]).Should().Contain(AgentHttpHeaderNames.ContainerId, expectedContainedId);
```

Reply:
I agree on the fluent assertions (there is also a `Assert.Equal` above), but I think `agent.RequestHeaders.Should().ContainSingle()` will still fail because `agent.RequestHeaders` should contain several headers in addition to `Datadog-Container-ID`.
https://github.com/DataDog/dd-trace-dotnet/blob/6f425500fb0ff4c03042995ab4f74336531289ed/tracer/src/Datadog.Trace/AgentHttpHeaderNames.cs#L74-L82

Reply:
Also, on Windows, the current test now asserts that the header is present with a `null` value. The tracer does not add the header at all on Windows. The previous code relies on `args.Value.Request.Headers[AgentHttpHeaderNames.ContainerId]` returning `null` when the header was missing.

Reply:
@lucaspimentel On your first point - yeah, this is pretty confusing. That's not quite how it works in the current implementation. `agent.RequestHeaders` is actually an `IImmutableList<NameValueCollection>` containing the headers of all the requests that have come in, but only the requests that came in including spans (which is why I renamed it to `TracerRequestHeaders`. Perhaps a better name could be `HeadersOfIncomingRequests` or `HeadersOfIncomingTraceRequests`, but I'm not sure a better name would quite solve this confusion. Do you have a better idea?

Reply:
> more importantly I think agent.RequestHeaders.Should().ContainSingle() will still fail because agent.RequestHeaders should contain several headers, 

You're right, this should fail I believe, but not for the reason you're saying 😄 agent.RequestHeaders is a list of lists, one list for each request. So we shouldn't be asserting that the outer list has a single request (because we may have 2 requests thanks to the "agent ping" at the start), but both of these requests _should_ contain the containerID (where applicable)

Agreed on the `null` behaviour too!

Reply:
@andrewlock I was attempting to simplify things by leveraging the fact that currently, we're only appending to the `RequestHeaders` list when we get traces (apparently), so the "agent ping" theoretically would not factor it?

Reply:
> the current test now asserts that the header is present with a null value, but the tracer does not add the header at all on Windows. The previous code relied on args.Value.Request.Headers[AgentHttpHeaderNames.ContainerId] returning null when the header was missing.

Fixed in 95739c93db893556b0ae72a955b2978424fa9ccd

Reply:
> so the "agent ping" theoretically would not factor into it? 

Unfortunately, the "agent ping" sends a trace request 😃 https://github.com/DataDog/dd-trace-dotnet/blob/6f425500fb0ff4c03042995ab4f74336531289ed/tracer/src/Datadog.Trace/Agent/AgentWriter.cs#L109-L112

> I renamed it to TraceRequestHeaders to be consistent with the pre-existing TelemetryRequestHeaders property.

Yep, I like the rename 👍